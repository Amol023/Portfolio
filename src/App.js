import React, { Component } from "react";
import { ThemeProvider } from "emotion-theming";
import Header from "../src/components/Header";
import Body from "../src/components/Body";
import Footer from "../src/components/Footer";
import styled, { css } from "react-emotion";
import theme from "./themes";

const flexify = props => {
  const {
    flexDirection = "row",
    justifyContent = "flex-start",
    flexFlow,
    alignItems = "flex-start"
  } = props;
  return css({
    flexDirection,
    justifyContent,
    flexFlow,
    alignItems
  });
};

export const BioContainer = styled("div")`
  ${flexify({ flexDirection: "row", justifyContent: "space-between" })};
  height: 100vh;
  border: 2px solid ${({ theme: { colors } }) => colors.grey.charcoal};
  border-radius: 2px;
`;

class App extends Component {
  render() {
    return (
      <ThemeProvider theme={theme}>
        <BioContainer>
          <Header />
          <Body />
          <Footer />
        </BioContainer>
      </ThemeProvider>
    );
  }
}

export default App;
