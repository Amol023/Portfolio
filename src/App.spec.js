import React from "react";
import * as emotion from "emotion";
import { createSerializer } from "jest-emotion";
import App from "./App";
import { getTestComponent } from "../test/helpers";

expect.addSnapshotSerializer(createSerializer(emotion));

describe("<App />", () => {
  it("renders without crashing", () => {
    const component = <App />;
    expect(getTestComponent({ component, mount: true })).toMatchSnapshot();
  });
});
