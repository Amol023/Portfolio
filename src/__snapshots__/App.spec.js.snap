// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders without crashing 1`] = `
.emotion-3 {
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  height: 100vh;
  border: 2px solid #404040;
  border-radius: 2px;
}

.emotion-0 {
  background: #009688;
  width: 100%;
  height: 5%;
  text-align: center;
}

.emotion-1 {
  width: 100%;
  height: 90%;
  text-align: center;
}

.emotion-2 {
  background: #7A7A7A;
  width: 100%;
  height: 5%;
  text-align: center;
}

<ThemeProvider
  theme={
    Object {
      "colors": Object {
        "black": Object {
          "jet": "#000000",
          "primary": "#222222",
        },
        "blue": Object {
          "light": "#6DA2B7",
        },
        "green": Object {
          "dark": "#0F5357",
          "teal": "#009688",
        },
        "grey": Object {
          "charcoal": "#404040",
          "dark": "#222222",
          "darkish": "#2E2E2E",
          "eclipse": "#373737",
          "faded": "#7A7A7A",
          "fadeout": "rgba(255, 255, 255, 0.95)",
          "hover": "#BBBBBB",
          "light": "#E5E5E5",
          "lightish": "#E9E9E9",
          "medDark": "#C4C4C4",
          "medium": "#DDDDDD",
          "semiDark": "#535353",
          "superlight": "#F0F0F0",
        },
      },
    }
  }
>
  <App>
    <ThemeProvider
      theme={
        Object {
          "colors": Object {
            "black": Object {
              "jet": "#000000",
              "primary": "#222222",
            },
            "blue": Object {
              "light": "#6DA2B7",
            },
            "green": Object {
              "dark": "#0F5357",
              "teal": "#009688",
            },
            "grey": Object {
              "charcoal": "#404040",
              "dark": "#222222",
              "darkish": "#2E2E2E",
              "eclipse": "#373737",
              "faded": "#7A7A7A",
              "fadeout": "rgba(255, 255, 255, 0.95)",
              "hover": "#BBBBBB",
              "light": "#E5E5E5",
              "lightish": "#E9E9E9",
              "medDark": "#C4C4C4",
              "medium": "#DDDDDD",
              "semiDark": "#535353",
              "superlight": "#F0F0F0",
            },
          },
        }
      }
    >
      <Styled(div)>
        <div
          className="emotion-3"
        >
          <Header>
            <Styled(div)>
              <div
                className="emotion-0"
              />
            </Styled(div)>
          </Header>
          <Body>
            <Styled(div)>
              <div
                className="emotion-1"
              />
            </Styled(div)>
          </Body>
          <Footer>
            <Styled(div)>
              <div
                className="emotion-2"
              />
            </Styled(div)>
          </Footer>
        </div>
      </Styled(div)>
    </ThemeProvider>
  </App>
</ThemeProvider>
`;
